ibis/expr/types/core.py

```diff
@@ -77,6 +84,7 @@ def has_name(self):
     def get_name(self):
         return self.op().resolve_name()
 
+    # TODO(kszucs): remove it entirely
```

Comment:
What's blocking this?

Reply:
Nothing really, it's a matter of backward compatibility. 

I have a follow-up patch to clearly mark named nodes, we can remove it there https://github.com/ibis-project/ibis/pull/4219/commits/cd982c75eb77b4b33e30e5d0bf2e7db5d1a14510